raven/gis.py

```diff
@@ -0,0 +1,472 @@
+        try:
+            if shape.endswith('shp'):
+                ogr.GetDriverByName('ESRI Shapefile')
+            return ogr.Open(shape)
+        except Exception as e:
+            msg = 'Failed to load Shapefile/GML: {}'.format(e)
+            raise Exception(msg)
+
+    @staticmethod
+    def _raster_counts(raster, band=0):
+        """A method for returning the cell values and their frequencies in a raster image."""
+        array = gdal_array.LoadFile(raster)
+        hist = array[band]
+
+        return Counter(hist.flatten())
+
+    @staticmethod
+    def _check_crs(shape=None, raster=None, crs=None):
+        """
+        Ensures that the CRS of the vector/raster and the specified CRS from EPSG string or shape/dem are compatible.
+        Needed to circumvent problems that can arise from OGC-compliant WKT and ESRI-WKT CRS strings.
+        Not sure of a better way to handle this issue.
+        """
+        try:
+            if shape is None and raster is None:
+                raise Exception("Please specify a shape or raster instance.")
+
+            elif all([shape, raster, crs]):
+                if len({shape, raster, crs}) == 1:
+                    return True
+                else:
+                    return False
+
+            elif (shape and raster and crs is None) or (shape and crs and raster is None) \
+                or (raster and crs and shape is None):
+                if shape == raster:
+                    return True
+                elif shape == crs:
+                    return True
+                elif raster == crs:
+                    return True
+                else:
+                    return False
+            else:
+                raise Exception("Please provide more arguments.")
+
+        except Exception as e:
+            msg = 'Failed to read Shape/DEM/CRS: {}'.format(e)
+            raise Exception(msg)
+
+
+class RavenRaster(RavenGIS):
+    """
+    Subclass of RavenGIS object
+
+    A GIS object initialized with a raster/DEM. Prevents coordinate transformations of the DEM/raster
+    which is important for discrete value categories (e.g. Land Use and Land Use Change data).
+    """
+
+    def __init__(self, raster):
+        if isinstance(raster, str):
+            super(RavenRaster, self).__init__()
+            self._raster = self.open_raster(raster, copy=None)
+            self._filename = os.path.basename(raster)
+            self._filepath = raster
+            self._pretty_crs, self._epsg, self._geo = self.extract_crs(raster=raster)
+        else:
+            raise Exception('File cannot be read!')
+
+    def __repr__(self):
+        return 'RavenGIS({!r}, {!r})'.format(self.__class__.__name__, self._filename)
+
+    def __str__(self):
+        return 'RavenGIS Raster: {}'.format(self._filename)
+
+    def projected(self):
+        if self._raster_reproj is not None:
+            return self._raster_reproj
+        return False
+
+    def stats(self):
```

Comment:
Stats should probably return a dict rather than print to stdout.

Reply:
I mean, you could rename this function to print_stats, which would call stats.

Reply:
I actually like the first option better. I'm not a fan of printing directly within a class either. I'll get to this soon.